include/swift/AST/TypeRepr.h

```diff
@@ -613,38 +613,38 @@ class NamedTypeRepr : public TypeRepr {
 
 /// \brief A protocol composite type.
 /// \code
-///   protocol<Foo, Bar>
+///   Foo & Bar
 /// \endcode
 class ProtocolCompositionTypeRepr : public TypeRepr {
   ArrayRef<IdentTypeRepr *> Protocols;
-  SourceLoc ProtocolLoc;
-  SourceRange AngleBrackets;
+  SourceLoc FirstTypeLoc;
+  SourceRange CompositionRange;
```

Comment:
This is indeed redundant — I have kept both so the old syntax uses the FirstTypeLoc as the protocol loc, and the range as the angle bracket range, for example for [this fixit](https://github.com/apple/swift/pull/3293/files#diff-a5731a934e0c085bf4b84dabd27a6ffdR2436). I would hope to change this once we remove support for the old synax and do not need this fixit.